### PR TITLE
Added get_tape_failure and delete_tape_failure (JAVACLI-65)

### PR DESCRIPTION
**Pull request by berkeleysquare**

### CODE REVIEW

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetTapeFailureView.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        return "" + result.getDetailedTapeFailures().size() + " Tape Failures:\n" +
+                ASCIITable.getInstance().getTable(getHeaders(), formatBucketList(result));
+    }
+
+    private String[][] formatBucketList(final DetailedTapeFailureList result) {
+        final List<DetailedTapeFailure> failures = result.getDetailedTapeFailures();
+        final String [][] formatArray = new String[failures.size()][];
+        int i = 0;
+        for(DetailedTapeFailure failure : failures) {
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetTapeFailureView.java]: Missing final

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetTapeFailureView.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import static com.spectralogic.ds3cli.util.Utils.nullGuard;
+
+public class GetTapeFailureView implements View<GetTapeFailureResult> {
+    @Override
+    public String render(final GetTapeFailureResult obj) {
+        final DetailedTapeFailureList result = obj.getResult();
+        if( (result == null)
+                || (null == result.getDetailedTapeFailures())
+                || (result.getDetailedTapeFailures().size() == 0) ) {
+            return "No taoe failures on remote appliance";
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetTapeFailureView.java]: `taoe` should be `tape`

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetTapeFailure.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import com.spectralogic.ds3client.models.Contents;
+import com.spectralogic.ds3client.networking.FailedRequestException;
+import com.spectralogic.ds3cli.models.GetTapeFailureResult;
+import org.apache.commons.cli.MissingOptionException;
+
+import java.util.Iterator;
+
+
+public class GetTapeFailure extends CliCommand<GetTapeFailureResult> {
+    private String bucketName;
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetTapeFailure.java]: Are these two variables being used?  If not they should be removed.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Ds3Cli.java [hunk 4]
```diff
@@ -193,8 +197,17 @@ private CliCommand getCommandExecutor() {
...
+                return new GetTapeFailure(this.ds3Provider, this.fileUtils);
+            }
+            case DELETE_TAPE_FAILURE: {
+                return new DeleteTapeFailure(this.ds3Provider, this.fileUtils);
+            }
+            case GET_SERVICE: {
+                return new GetService(this.ds3Provider, this.fileUtils);
+            }
             default: {
+                System.err.println("Command not recognized");
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Ds3Cli.java]: I would log this.  And most likely we should end up changing this to return an exception, but we should do that work in a different pull request.

Comment (berkeleysquare) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Ds3Cli.java]: I didn't know if there was appetite, but I wondered about registering commands with Ds3Cli and getting rid of the long if then stream. This is a funny error because if you type a bad command it will be caught by the argument parser. This is only hit when you have one defined in CommandType but not implemented. Old behavior was to default to getService() but I can attest to that's not being terribly intuitive.
